pandas/core/internals/managers.py

```diff
@@ -430,7 +430,10 @@ def apply(
                     axis = obj._info_axis_number
                     kwargs[k] = obj.reindex(b_items, axis=axis, copy=align_copy)
 
-            applied = getattr(b, f)(**kwargs)
+            if callable(f):
```

Comment:
Should we move this type of logic to apply? I think would be more consistent with user-facing parts of the API

Reply:
this is in BlockManager.apply.  Do you mean move it to Block.apply?  I guess that'd be fine.  ATM we only have a handful of functions that call BlockManager.apply (xref #29825)